Expensify/App
# Add VBA flow Part 1

### Details

Adds the Verified Bank Account flow to E.cash

For this PR I am just cleaning things up a bit and laying down some scaffolding. We will:

- [x] Create a new route `/bank-account/new` for adding the VBA
- [x] Extract the Plaid logic we need from the PBA flow so it can be used in both views
- [x] Add the view for manually adding a bank account

We will not be hooking up to any APIs just yet...

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/166041

### Tests
1. Navigate to `/bank-account/new`
2. Verify the add method selection options are shown
3. Select manual
4. Verify you can only press the button when valid account number + routing number are entered
5. Select Log Into Bank
6. Verify the Plaid widget appears
7. Enter chase bank info `user_good` `pass_good` + Continue
8. Verify the options are shown in a Picker
9. Select an option and enter password
10. Repeat Plaid steps for `/add-personal-bank-account` route and verify the selection screen appears + an account can be added (confirm via Web-Expensify).

**Note:** Testing on native/desktop is tricky atm but can be done by using a deep link for native and/or putting the `navigate()` method on window like this:

```diff
diff --git a/src/libs/Navigation/Navigation.js b/src/libs/Navigation/Navigation.js
index 7fe26fcf3..0abf6ddee 100644
--- a/src/libs/Navigation/Navigation.js
+++ b/src/libs/Navigation/Navigation.js
@@ -66,6 +66,8 @@ function navigate(route = ROUTES.HOME) {
     linkTo(navigationRef.current, route);
 }

+window.ecashNavigate = navigate;
+
 /**
  * Dismisses a screen presented modally and returns us back to the previous view.
  *
```

then calling `window.ecashNavigate('/bank-account/new')` from console / native debugger session

### QA Steps
No QA - We will test the entire flow together at the end

### Tested On


### Screenshots


#### Web
![2021-06-09_13-12-28](https://user-images.githubusercontent.com/32969087/121441013-b35aa580-c924-11eb-9662-52156dca104a.png)
![2021-06-09_13-12-32](https://user-images.githubusercontent.com/32969087/121441021-b6559600-c924-11eb-99f7-8be5cb2c4c48.png)
![2021-06-09_13-12-44](https://user-images.githubusercontent.com/32969087/121441026-b8b7f000-c924-11eb-95ac-e26315722607.png)
![2021-06-09_13-12-59](https://user-images.githubusercontent.com/32969087/121441031-bc4b7700-c924-11eb-8057-9a7b111f1c85.png)
![2021-06-09_13-13-10](https://user-images.githubusercontent.com/32969087/121441035-be153a80-c924-11eb-9e4e-67fc66f10b49.png)
![2021-06-09_13-13-16](https://user-images.githubusercontent.com/32969087/121441046-c1a8c180-c924-11eb-9730-7dd8d3f4ba25.png)
![2021-06-09_13-13-20](https://user-images.githubusercontent.com/32969087/121441059-c4a3b200-c924-11eb-812a-e8b2bac3b9a4.png)
![2021-06-09_13-13-26](https://user-images.githubusercontent.com/32969087/121441065-c79ea280-c924-11eb-98d5-d95efbafa350.png)
![2021-06-09_13-13-30](https://user-images.githubusercontent.com/32969087/121441072-ca00fc80-c924-11eb-9dc8-8067716b897c.png)

#### Mobile Web
![2021-06-09_13-17-13](https://user-images.githubusercontent.com/32969087/121441937-5f50c080-c926-11eb-8c28-86e21d13bfa7.png)
![2021-06-09_13-17-17](https://user-images.githubusercontent.com/32969087/121441941-611a8400-c926-11eb-903b-b1b8bd60eb31.png)
![2021-06-09_13-24-35](https://user-images.githubusercontent.com/32969087/121441944-62e44780-c926-11eb-9e8a-da41add1aef8.png)
![2021-06-09_13-24-49](https://user-images.githubusercontent.com/32969087/121441946-64157480-c926-11eb-8dc6-a0c3d3519684.png)
![2021-06-09_13-24-59](https://user-images.githubusercontent.com/32969087/121441950-6546a180-c926-11eb-95cb-b973254862ab.png)
![2021-06-09_13-25-07](https://user-images.githubusercontent.com/32969087/121441955-6677ce80-c926-11eb-97d4-329218938098.png)
![2021-06-09_13-25-33](https://user-images.githubusercontent.com/32969087/121441963-6972bf00-c926-11eb-8c92-d1d94f2b8eb7.png)
![2021-06-09_13-25-33](https://user-images.githubusercontent.com/32969087/121441969-6c6daf80-c926-11eb-8fd1-96ecaa6a84e7.png)
![2021-06-09_13-25-37](https://user-images.githubusercontent.com/32969087/121441974-6d9edc80-c926-11eb-91b2-d7fa11e87440.png)
![2021-06-09_13-25-41](https://user-images.githubusercontent.com/32969087/121441978-6f68a000-c926-11eb-8044-6eff934d1e54.png)
![2021-06-09_13-25-52](https://user-images.githubusercontent.com/32969087/121441983-71326380-c926-11eb-99b3-e6ebfb6ff36b.png)

#### Desktop
![2021-06-09_14-35-45](https://user-images.githubusercontent.com/32969087/121446915-4e0cb180-c930-11eb-99f7-606d159d3495.png)
![2021-06-09_14-35-56](https://user-images.githubusercontent.com/32969087/121446919-4fd67500-c930-11eb-9b44-763b12747515.png)
![2021-06-09_14-36-15](https://user-images.githubusercontent.com/32969087/121446926-5107a200-c930-11eb-8ecf-3c3f8b43b4ae.png)
![2021-06-09_14-36-21](https://user-images.githubusercontent.com/32969087/121446931-52d16580-c930-11eb-92ec-8241ddc83b73.png)
![2021-06-09_14-37-08](https://user-images.githubusercontent.com/32969087/121446934-549b2900-c930-11eb-987b-756e0c57389c.png)

#### iOS
![2021-06-09_13-28-40](https://user-images.githubusercontent.com/32969087/121442653-cde24e00-c927-11eb-9f3d-e19644a27f5e.png)
![2021-06-09_13-28-53](https://user-images.githubusercontent.com/32969087/121442656-cfac1180-c927-11eb-8704-ae3e661d07d6.png)
![2021-06-09_13-29-08](https://user-images.githubusercontent.com/32969087/121442660-d0dd3e80-c927-11eb-96e2-1aeb2c839ff2.png)
![2021-06-09_13-29-14](https://user-images.githubusercontent.com/32969087/121442665-d33f9880-c927-11eb-878f-7c20ef47cdd6.png)
![2021-06-09_13-29-19](https://user-images.githubusercontent.com/32969087/121442676-d5a1f280-c927-11eb-81f2-9afd146a2c4a.png)
![2021-06-09_13-29-29](https://user-images.githubusercontent.com/32969087/121442682-d8044c80-c927-11eb-9fbe-46fd621c0052.png)
![2021-06-09_13-32-51](https://user-images.githubusercontent.com/32969087/121442692-dcc90080-c927-11eb-9c93-356e1906970a.png)
![2021-06-09_13-36-42](https://user-images.githubusercontent.com/32969087/121442694-df2b5a80-c927-11eb-8bde-865020d895f6.png)

#### Android

![2021-06-09_13-56-04](https://user-images.githubusercontent.com/32969087/121444346-0cc5d300-c92b-11eb-8406-e55691879540.png)
![2021-06-09_13-57-34](https://user-images.githubusercontent.com/32969087/121444350-0fc0c380-c92b-11eb-8383-d54901e5a686.png)
![2021-06-09_13-57-56](https://user-images.githubusercontent.com/32969087/121444357-12231d80-c92b-11eb-8d1a-dc5c84a790a1.png)
![2021-06-09_13-58-40](https://user-images.githubusercontent.com/32969087/121444364-164f3b00-c92b-11eb-858c-cf313f564092.png)
![2021-06-09_13-59-17](https://user-images.githubusercontent.com/32969087/121444368-1818fe80-c92b-11eb-9538-7f8ab02e5b3c.png)
![2021-06-09_13-59-17](https://user-images.githubusercontent.com/32969087/121444376-1b13ef00-c92b-11eb-8361-d2ecb8f46fb7.png)
![2021-06-09_13-59-23](https://user-images.githubusercontent.com/32969087/121444378-1c451c00-c92b-11eb-89d6-d0bc9c35c5a4.png)
![2021-06-09_13-59-41](https://user-images.githubusercontent.com/32969087/121444381-1ea77600-c92b-11eb-89fd-fd348335cc19.png)
